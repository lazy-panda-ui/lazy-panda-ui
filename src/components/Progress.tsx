import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { useTheme, Theme } from '../theme';

export interface ProgressProps {
  value: number; // 0 to 1
  style?: ViewStyle;
}

export const Progress: React.FC<ProgressProps> = ({ value, style }) => {
  const theme = useTheme();
  return (
    <View style={[styles(theme).container, style]}>
      <View style={[styles(theme).bar, { width: Math.max(0, Math.min(1, value)) * 100 + '%' as unknown as number }]} />
    </View>
  );
};

const styles = (theme: Theme) => StyleSheet.create({
  bar: {
    backgroundColor: theme.colors.primary,
    borderRadius: 4,
    height: 8,
  },
  container: {
    backgroundColor: theme.colors.border,
    borderRadius: 4,
    height: 8,
    overflow: 'hidden',
    width: '100%',
  },
});
