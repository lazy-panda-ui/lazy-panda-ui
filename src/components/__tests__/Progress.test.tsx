import React from 'react';
import { render } from '@testing-library/react-native';
import { Progress } from '../Progress';

describe('Progress', () => {
  it('renders with value', () => {
    const { toJSON } = render(<Progress value={0.5} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
