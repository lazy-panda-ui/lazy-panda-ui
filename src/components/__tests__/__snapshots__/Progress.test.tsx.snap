// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders with value 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#E5E5EA",
        "borderRadius": 4,
        "height": 8,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#007AFF",
          "borderRadius": 4,
          "height": 8,
        },
        {
          "width": "50%",
        },
      ]
    }
  />
</View>
`;
